libraries/AP_Camera/AP_Camera.h

```diff
@@ -113,11 +113,16 @@ class AP_Camera {
 
     // take a picture
     void take_picture();
-    void take_picture(uint8_t instance);
+    bool take_picture(uint8_t instance);
 
     // take multiple pictures, time_interval between two consecutive pictures is in miliseconds
     // total_num is number of pictures to be taken, -1 means capture forever
     void take_multiple_pictures(uint32_t time_interval_ms, int16_t total_num);
+    bool take_multiple_pictures(uint8_t instance, uint32_t time_interval_ms, int16_t total_num);
```

Comment:
I think we should make these take-multiple-pictures methods fail if the total_num is non-zero (e.g. forever or some specified number) and the time_interval_ms is zero.  I know you had this originally in the command parser and I asked you to remove it but after testing I see we really do need the check of a non-zero interval... it is better to have it down here in this method anyway though.

Reply:
Okay upto you then i have rebased